girder/api/access.py

```diff
@@ -80,3 +77,15 @@ def accessDecorator(*args, **kwargs):
         return fun(*args, **kwargs)
     accessDecorator.accessLevel = 'token'
     return accessDecorator
+
+
+def cookie(fun):
+    """
+    REST endpoints that allow the use of a cookie for authentication should be
+    wrapped in this decorator.
+
+    This should only be set on endpoints that are read-only (GET requests),
+    to avoid vulnerabilities to Cross-Site Request Forgery attacks.
```

Comment:
@cpatrick good idea: ``@cookie`` or ``@cookie(force=True)``, allowing only GET or HEAD by default.

It significantly complicates the internal implementation to have a decorator optionally take a parameter, but that's not a problem.